src/bson.c

```diff
@@ -137,6 +137,14 @@ static bool php_phongo_bson_visit_binary(const bson_iter_t *iter ARG_UNUSED, con
 	return false;
 } /* }}} */
 
+static bool php_phongo_bson_visit_undefined(const bson_iter_t *iter, const char *key, void *data) /* {{{ */
+{
+	TSRMLS_FETCH();
+	phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Detected unsupported BSON type 0x06 (undefined) for fieldname \"%s\"", key);
+
+	return true;
```

Comment:
Return `false` here so that BSON visitation can continue and we can report multiple exceptions for a single document. Subsequent exceptions will be accessible via `getPrevious()`.